integration/simulation/simulation_in_mem_test.go

```diff
@@ -24,11 +36,78 @@ func TestInMemoryMonteCarloSimulation(t *testing.T) {
+
+	tmpClient := buildhelper.NewEthAPI("127.0.0.1", 7545)
+	err := tmpClient.Connect()
+	if err != nil {
+		panic(err)
+	}
+
+	deployContractTx := types.LegacyTx{
+		Nonce:    0, // relies on a clean env
```

Comment:
@otherview Can't we just set the nonce dynamically, allowing us to rerun the sim?

    nonce, err := tmpClient.ApiClient.PendingNonceAt(context.Background(), buildhelper.Addr1())

Reply:
Not really since the validations of the simulation expect an empty run.

So stuff like counting blocks and rollups and the number of deposits start to lose meaning when you have 2 or more executions in the same ganache.